pkg/appgw/backendaddresspools.go

```diff
@@ -7,82 +7,121 @@ package appgw
-				// The same service might be referenced in multiple ingress resources, this might result in multiple `serviceBackendPairMap` having the same service key but different
-				// ingress resource. Thus, while generating the backend address pool, we should make sure that we are generating unique backend address pools.
-				addressPool, ok := addressPools[addressPoolName]
+func getBackendPoolMapValues(m *map[string]*n.ApplicationGatewayBackendAddressPool) *[]n.ApplicationGatewayBackendAddressPool {
+	var backendAddressPools []n.ApplicationGatewayBackendAddressPool
+	for _, addr := range *m {
+		backendAddressPools = append(backendAddressPools, *addr)
+	}
+	return &backendAddressPools
+}
 
-				if !ok {
-					addressPoolAddresses := make([](network.ApplicationGatewayBackendAddress), 0)
-					for _, address := range subset.Addresses {
-						ip := address.IP
-						hostname := address.Hostname
-						// prefer IP address
-						if len(ip) != 0 {
-							// address specified by ip
-							addressPoolAddresses = append(addressPoolAddresses, network.ApplicationGatewayBackendAddress{IPAddress: &ip})
-						} else if len(address.Hostname) != 0 {
-							// address specified by hostname
-							addressPoolAddresses = append(addressPoolAddresses, network.ApplicationGatewayBackendAddress{Fqdn: &hostname})
-						}
-					}
+func getBackendAddressMapKeys(m *map[n.ApplicationGatewayBackendAddress]interface{}) *[]n.ApplicationGatewayBackendAddress {
+	var addresses []n.ApplicationGatewayBackendAddress
+	for addr := range *m {
+		addresses = append(addresses, addr)
+	}
+	sort.Sort(byIPFQDN(addresses))
+	return &addresses
+}
 
-					addressPool = network.ApplicationGatewayBackendAddressPool{
-						Etag: to.StringPtr("*"),
-						Name: &addressPoolName,
-						ApplicationGatewayBackendAddressPoolPropertiesFormat: &network.ApplicationGatewayBackendAddressPoolPropertiesFormat{
-							BackendAddresses: &addressPoolAddresses,
-						},
-					}
+func getPorts(subset v1.EndpointSubset) map[int32]interface{} {
+	ports := make(map[int32]interface{})
+	for _, endpointsPort := range subset.Ports {
+		if endpointsPort.Protocol == v1.ProtocolTCP {
+			ports[endpointsPort.Port] = nil
+		}
+	}
+	return ports
+}
 
-					addressPools[*addressPool.Name] = addressPool
-				}
+func (builder *appGwConfigBuilder) getBackendAddressPool(backendID backendIdentifier, serviceBackendPair serviceBackendPortPair, defaultPool *n.ApplicationGatewayBackendAddressPool, addressPools *map[string]*n.ApplicationGatewayBackendAddressPool) *n.ApplicationGatewayBackendAddressPool {
+	endpoints := builder.k8sContext.GetEndpointsByService(backendID.serviceKey())
+	if endpoints == nil {
+		logLine := fmt.Sprintf("Unable to get endpoints for service key [%s]", backendID.serviceKey())
+		builder.recorder.Event(backendID.Ingress, v1.EventTypeWarning, "EndpointsEmpty", logLine)
+		glog.Warning(logLine)
+		return defaultPool
```

Comment:
`return nil` here to be more explicit that this function did nothing ?

Reply:
Reading the code how it was before the refactor - we want to return the `defaultPool` (what was formerly `emptyPool`).  Here is where this is coming from:  https://github.com/Azure/application-gateway-kubernetes-ingress/blob/a495cc5d2674c1bf18b46574e3e6631824a776a7/pkg/appgw/backendaddresspools.go#L23-L29